src/azure-cli/azure/cli/command_modules/storage/_client_factory.py

```diff
@@ -220,6 +220,13 @@ def cf_blob_client(cli_ctx, kwargs):
                                                             snapshot=kwargs.pop('snapshot', None))
 
 
+def cf_blob_lease_client(cli_ctx, kwargs):
+    t_lease_service = get_sdk(cli_ctx, ResourceType.DATA_STORAGE_BLOB, '_lease#BlobLeaseClient')
+    blob_client = cf_blob_service(cli_ctx, kwargs).get_blob_client(container=kwargs.pop('container_name', None),
```

Comment:
>pop [](start = 84, length = 3)

why do we need `pop` (remove) from kwargs?

Reply:
discussed offline